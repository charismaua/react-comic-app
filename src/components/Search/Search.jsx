import React from "react";
import PropTypes from "prop-types";
import styled from "@emotion/styled";
import { lighten, darken } from "polished";

const StyledSearch = styled.div`
  width: 100%;
  background-color: #114b88;

  .search__wrapper {
    max-width: 450px;
    display: flex;
    justify-content: space-between;
    width: 100%;
    padding: 20px;
    margin: 0 auto;
  }

  .search__form-element {
    height: 48px;
    padding: 0 20px;
    border-style: solid;
    border-width: 1px;
    line-height: 1;
    font-weight: normal;
    outline: none;
    border-radius: 2px;
    font-size: 16px;
  }

  .search__input {
    width: 100%;
    border-color: #d9dce7;
    color: #141432;
    background-color: #fff;
    transition: box-shadow 0.25s ease-out;
    margin-right: 20px;

    &:focus {
      box-shadow: 0 0 10px #57a3ff;
    }
  }

  .search__button {
    border: 1px solid #3590cf;
    color: #fff;
    background-color: #3590cf;
    transition: background-color 0.25s ease-out;
    cursor: pointer;

    &:hover,
    &:focus {
      background-color: ${lighten(0.05, "#3590cf")};
    }

    &:active {
      background-color: ${darken(0.05, "#3590cf")};
    }
  }

  .search__form {
    display: flex;
    width: 100%;
  }
`;

const Search = ({ onSubmit }) => {
  return (
    <StyledSearch data-testid="Search">
      <div className="search__wrapper">
        <form className="search__form" onSubmit={onSubmit}>
          <input
            className="search__form-element search__input"
            type="number"
            placeholder="Type the comic's number here..."
          />
          <button type="submit" className="search__form-element search__button">
            Submit
          </button>
        </form>
      </div>
    </StyledSearch>
  );
};

Search.propTypes = {
  onSubmit: PropTypes.func,
};

Search.defaultProps = {
  onSubmit: null,
};

export default Search;
