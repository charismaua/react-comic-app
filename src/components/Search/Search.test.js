import { render } from '@testing-library/react';
import Search from './Search';

describe('Search', () => {
    it('should render component without throwing an error', async () => {
        const { container, queryByTestId } = render(<Search />);
        expect(queryByTestId('Search')).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it('should render component with proper width', async () => {
        const { container, getAllByTestId } = render(<Search />);
        const style = window.getComputedStyle(getAllByTestId('Search')[0]);

        expect(style.width).toBe('100%');
        expect(container).toMatchSnapshot();
    });
});