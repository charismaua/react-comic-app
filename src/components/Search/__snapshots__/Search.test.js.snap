// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render component with proper width 1`] = `
<div>
  <div
    class="css-bf2dn0"
    data-testid="Search"
  >
    <div
      class="search__wrapper"
    >
      <form
        class="search__form"
      >
        <input
          class="search__form-element search__input"
          placeholder="Type the comic's number here..."
          type="number"
        />
        <button
          class="search__form-element search__button"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`Search should render component without throwing an error 1`] = `
<div>
  <div
    class="css-bf2dn0"
    data-testid="Search"
  >
    <div
      class="search__wrapper"
    >
      <form
        class="search__form"
      >
        <input
          class="search__form-element search__input"
          placeholder="Type the comic's number here..."
          type="number"
        />
        <button
          class="search__form-element search__button"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</div>
`;
